README.md

```diff
@@ -168,6 +169,63 @@ Make sure that you have your Home Assistant Container network set to 'host', as
 
 - [Google Home timers card](https://github.com/DurgNomis-drol/google_home_timers_card) by [@DurgNomis-drol](https://github.com/DurgNomis-drol)
 
+## Troubleshooting
+
+### Collecting useful log data
+
+Here are the steps to generate useful log data:
+
+1. Temporary log level change.
+    1. Visit [![Open your Home Assistant instance and show your service developer tools.](https://my.home-assistant.io/badges/developer_services.svg)](https://my.home-assistant.io/redirect/developer_services/)
+    2. Choose `Logger: Set level` from the **Service** menu.
+    3. Under **Service data** paste the following:
+        ```yaml
+        custom_components.google_home: debug
+        glocaltokens: debug
+        ```
+    4. Click **Call Service**.
+2. Read the log information.
+    1. Visit [![Open your Home Assistant instance and show your Home Assistant logs.](https://my.home-assistant.io/badges/logs.svg)](https://my.home-assistant.io/redirect/logs/)
+    2. Click **Load Full Home Assistant Log**.
+    3. Look for all `google_home` entries.
+    4. Also look for all `glocaltokens` entries.
+3. Requesting help with the log information.
+    1. Copy the log entries.
+    2. Paste them into a discussion forum or bug report between the following two lines:
+        >\```
+        >
+        >\```
+
+> Advanced tips:
+> * Adding ` ```python ` to the beginning of your code block instead of just ` ``` ` adds color coding to make the logs easier for developers to read.
+> * Use [LogExpert](https://github.com/zarunbal/LogExpert/) to make finding log entries quicker.
+>   1. Open home-assistant.log in your config folder.
+>   2. Use `glocaltokens|google_home` as your _Text filter_ to include logs from both _glocaltokens_ and _google_home_.
+
+### "Username/Password is incorrect"
+If you get this error:
+1. First verify you are using the correct Username and Password combination for that Google account.
+2. Have you enabled 2 Factor Authentication on thant Google account? _If so read the [2 Factor Authentication](#2-factor-authentication--app-passwords) section to continue._
+3. After ruling out #1 and #2: _Read the [Collecting useful log data](#collecting-useful-log-data) section._
+
+### 2 Factor Authentication / App Passwords
+#### Error: "The setting you are looking for is not available for your account." at https://myaccount.google.com/apppasswords. 
+This error will occur if you do not have 2 Factor Authentication (2FA) setup on your Google account.
+
+  Here are the steps to correct this issue:
+  1. If you have multiple Google accounts go to [Google.com](https://google.com) and log out of every accout!
+  (_This causes a lot of confusion! You may think you have enabled 2FA only to be enabling it on an alternate Google account!_)
+  2. Go to [Google.com](https://google.com) and **only** log into the Google Home account you want to control with the Google Home Integration.
```

Comment:
Btw, I registered a separate google account just for this and shared access to Home with it. This way you can avoid providing sensitive access to the main account to this integration. Probably you can add a not about this.

Reply:
I have a Google Apps for my Domain account, which isn't compatible with all of Google's services (Youtube Premium for example), so I have to also keep a @gmail.com account for use with most Google services as well.